tests/integration/cqlengine/columns/test_validation.py

```diff
@@ -187,49 +184,173 @@ def test_varint_io(self):
-        result = self.DateTest.objects(test_id=0).first()
-        self.assertEqual(result.created_at, util.Date(today))
+    def _check_value_is_correct_in_db(self, value):
+
+        if value is None:
+            result = self.model_class.objects.all().allow_filtering().filter(test_id=0).first()
+            self.assertIsNone(result.class_param)
+
+        else:
+            if not isinstance(value, self.python_klass):
+                value_to_compare = self.python_klass(value)
+            else:
+                value_to_compare = value
+
+            result = self.model_class.objects(test_id=0).first()
+            self.assertIsInstance(result.class_param, self.python_klass)
+            self.assertEqual(result.class_param, value_to_compare)
+
+            result = self.model_class.objects.all().allow_filtering().filter(test_id=0).first()
+            self.assertIsInstance(result.class_param, self.python_klass)
+            self.assertEqual(result.class_param, value_to_compare)
+
+            result = self.model_class.objects.all().allow_filtering().filter(test_id=0, class_param=value).first()
+            self.assertIsInstance(result.class_param, self.python_klass)
+            self.assertEqual(result.class_param, value_to_compare)
+
+        return result
 
     def test_date_io_using_datetime(self):
-        now = datetime.utcnow()
-        self.DateTest.objects.create(test_id=0, created_at=now)
-        result = self.DateTest.objects(test_id=0).first()
-        self.assertIsInstance(result.created_at, util.Date)
-        self.assertEqual(result.created_at, util.Date(now))
+        first_value = self.first_value
+        second_value = self.second_value
+        third_value = self.third_value
+
+        self.model_class.objects.create(test_id=0, class_param=first_value)
+        result = self._check_value_is_correct_in_db(first_value)
+        result.delete()
+
+        self.model_class.objects.create(test_id=0, class_param=second_value)
+        result = self._check_value_is_correct_in_db(second_value)
+
+        result.update(class_param=third_value).save()
+        result = self._check_value_is_correct_in_db(third_value)
+
+        result.update(class_param=None).save()
+        self._check_value_is_correct_in_db(None)
 
     def test_date_none(self):
-        self.DateTest.objects.create(test_id=1, created_at=None)
-        dt2 = self.DateTest.objects(test_id=1).first()
-        assert dt2.created_at is None
+        self.model_class.objects.create(test_id=1, class_param=None)
+        dt2 = self.model_class.objects(test_id=1).first()
+        self.assertIsNone(dt2.class_param)
+
+        dts = self.model_class.objects(test_id=1).values_list('class_param')
+        self.assertIsNone(dts[0][0])
+
+
+class TestDate(DataType, BaseCassEngTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.db_klass, cls.python_klass = Date, util.Date
+        cls.first_value, cls.second_value, cls.third_value = \
```

Comment:
Oh, I think this wasn't addressed -- could you change this to iterate over an iterable of values, rather than hardcoding attributes for them?